### ISSUE TITLE
PHP directors code generation

### ISSUE THREAD

**Issue by tomterl** [swig/swig#1900]
I try to build an FLTK php extension using swig.  I seem to need to use directors to get the constructors to work correctly.  Doing so results in code where the generated C++ code and the php wrapper are at odds:
This declaration in the header file (FL/Fl_Widget.H)

````
/** Gets the font size in pixels.
      The default size is 14 pixels.
      \return the current font size
   */
  Fl_Fontsize labelsize() const {return label_.size;}

  /** Sets the font size in pixels.
      \param[in] pix the new font size
      \see Fl_Fontsize labelsize()
   */
  void labelsize(Fl_Fontsize pix) {label_.size=pix;}
````

produces two corresponding methods in the generated C++ (with names ending in SWIG_0 for the getter, and SWIG_1 for the setter), the setter returning void, the getter returning the current label size, an integer value.

The php-wrapper method does use both returned values (void and integer) as value for the internal object-pointer, rendering the object invalid (and probably leaking memory in the process, IDK), and returning an unset value `$r`:
````
function labelsize($pix=null) {
                switch (func_num_args()) {
                case 0: $this->_cPtr=Fl_Widget_labelsize($this->_cPtr); break;
                default: $this->_cPtr=Fl_Widget_labelsize($this->_cPtr,$pix);
                }
                return $r;
        }
````

If I manually change these functions (a lot of them are wrong) to look like this:
````
function labelsize($pix=null) {
                switch (func_num_args()) {
                case 0: $r=Fl_Widget_labelsize($this->_cPtr); break;
                default: $r=Fl_Widget_labelsize($this->_cPtr,$pix);
                }
                return $r;
        }
````
I at least don't get any errors and manage to display a frame on my desktop.

Looking at php.cxx, swigs php module, the `$this->_cPtr` assignement seems to happen when I have directors active, I'm not completly sure, the code is somewhat hard to follow at a glance (I'd assumed `$r=` to be used, because while `Fl_Widget` has virtual methods, `labelsize` isn't).

Is this an error in swig, or did I miss something in the code or documentation that could help me out?

kind regards,
Tom

**Comment by tomterl** [comment]
We have a workaround, or maybe this is how it is supposed to be used: we use %rename to deduplicate getter/setter functions; but my question remains: Shouldn't the generated xxx extension code and the php wrapper work without rename in this case?

**Comment by ojwb** [comment]
It's a bug (though it shouldn't leak memory as the old value of `_cPtr` should get garbage collected by PHP).

**Comment by tomterl** [comment]
I just build swig master with these two changes, and the ftlk binding test we have in place works fine without the need to `%rename`, as expected.

Thanks!
